### ISSUE TITLE
Spike: There is a way to properly edit column headers in the new decision table layout

### ISSUE THREAD

**Issue by pinussilvestrus** [bpmn-io/dmn-js#490]
__What should we do?__

After we investigated a [new decision table layout](https://github.com/bpmn-io/dmn-js/issues/477) which improves the understandability of decision logic, we should invest in a way on how to edit table column headers.

__Why should we do it?__

Given the following, new decision table layout ...

![image](https://user-images.githubusercontent.com/9433996/76748412-fe662700-677a-11ea-850e-136d147fb04c.png)

(Cf. https://dmn-prototypes.netlify.com/decision-table-layout/)

... there should be a way to edit column headers properly in
* setting the input/output name
* setting the input/output type
* creating new columns
* defining engine variable names

Additional stuff is ...
* changing the column order?
* predefined values?
* hiding the When/And/Then clauses?

__Resources__
* [DMN Hypothesis Board](https://docs.google.com/spreadsheets/d/1ZI1GDUHJFP_RWLeIKFFxzFAlyJzH4Ozt2ApZnHIhVqQ/edit#gid=0)
* [Summary Docs](https://docs.google.com/document/d/1kZ0maN6uNKNW49d-d6b1okJ6uWuPe1QYDNKQkKtHds4/edit#heading=h.lqr5z1j7vgl2)
* [`dmn-prototypes`](https://github.com/pinussilvestrus/dmn-prototypes)  
* [Demo](https://dmn-prototypes.netlify.com/column-editing/?variant=A1)

**Comment by pinussilvestrus** [comment]
![sketch_1](https://user-images.githubusercontent.com/9433996/77052053-8b53ef00-69cc-11ea-9994-27ebcab70b12.png)

![sketch_3](https://user-images.githubusercontent.com/9433996/77052058-8c851c00-69cc-11ea-8321-dbedbe86ebb7.png)

**Comment by pinussilvestrus** [comment]
One Approach:

![Kapture 2020-03-20 at 15 57 00](https://user-images.githubusercontent.com/9433996/77175816-7c039d00-6ac3-11ea-8447-64bfef3c4856.gif)

**Comment by pinussilvestrus** [comment]
> 🚧 Work in progress Summary

We came with the following concept:

* Direct Editing as edit component (variant B)
* Set a max-width for expressions per default and crop content with an ellipsis (variant B2)
* See full content in edit mode
* Make it possible to resize columns if the user needs it (variant B1)
* Only allow one-line expressions, no text wrap

TBA: Screencast of final Demo

__Reason__
* It makes way clearer what the user is actively editing (input/output expression), reduced noise in comparison to the Modal (variant A)
* We want the user to shorten its expressions. If he really needs it he is able to use the resize OR should go to literal expressions

--> That's the behavior user knows from a normal spreadsheet editor like Google Sheets

We won't do user tests for now as we are happy with the solution we found. We think about testing the implementation and iterate over it.
